### PR TITLE
Update Account page logic

### PR DESCRIPTION
**Pull request by hjiangsu**
## Pull Request Description

This draft PR updates our Account page logic to use the feed page under-the-hood. ~~This feature is currently only enabled when experimental features are on.~~ Migrating the account page to use the feed page under the hood provides us with a few major benefits:
- Overall improved performance, since we will be using slivers instead of the standard ListView builder
- Automatic hiding of user header bar (similar to user feed pages) as it uses the same logic as the feed page. This allows us to view more of our posts/comments on the screen
- Reduces code complexity and allows us to deprecate older legacy code that was kept for the old Account page
- Reduces the number of bugs due to missing blocs/context since it is all shared now. This allows us to perform all actions as we would on the user feed pages.
- Enables the ability to change the sorting of our posts/comments

There are a few changes that had to be made in order for this to work:
- When viewing the Account page, the app bar has been adjusted to show the current account page actions. The app bar also includes additional actions present in the general user feed pages (sharing user profile and sorting)
- The app bar now has a toggle to switch between saved posts/comments, and your own posts/comments. While it would be possible to bring back the old behaviour eventually, this is currently the simplest way to handle the logic for now without too many dependencies being created across widgets.

## Issue Being Fixed

As a side effect, this PR should also solve #1521!

Issue Number: #1521

## Screenshots / Recordings

Some screenshots of the new Account page.

| Posts/Comments | Saved Posts/Comments |
|-|-|
|Screenshot_1732076085 |Screenshot_1732076100|

## Checklist

- [ ] If a new package was added, did you ensure it uses an appropriate license and is actively maintained?
- [ ] Did you use localized strings (and added appropriate descriptions) where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

**Comment by micahmo** [comment]
Looking forward to this! 😊

**Comment by hjiangsu** [comment]
Alright, I believe I implemented most of the functionality now!

@micahmo feel free to try this out and let me know if you notice any issues or any missing features that were present in the previous implementation!

File: lib/account/widgets/account_page_app_bar.dart [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  final void Function(bool showSaved)? onToggleSaved;
+
+  @override
+  Widget build(BuildContext context) {
+    final l10n = AppLocalizations.of(context)!;
+    final feedBloc = context.read<FeedBloc>();
+
+    return Row(
+      children: [
+        IconButton(
```
Comment (micahmo) [inline lib/account/widgets/account_page_app_bar.dart:143]: Any chance you could add a tooltip here?

**Review by micahmo — APPROVED** [review]
Very nice!!

File: lib/account/widgets/account_page_app_bar.dart (hunk 1 above)
Comment (hjiangsu) [reply lib/account/widgets/account_page_app_bar.dart:143]: Done in https://github.com/thunder-app/thunder/pull/1597/commits/a736b9d0eff5dcfff5a689c9d993091c88fb2076!

**Pull request merged**